dragonfly/implementations/daq_run_interface.py

```diff
@@ -502,50 +491,67 @@ def _start_data_taking(self, directory, filename):
-                result = self.provider.cmd(self.psyllid_interface, 'start_run', payload=payload)
-
-            # if the run time exceeds the set max duration the run is split into sub acquisitions
-            else:
-                logger.info('Doing {} acquisitions'.format(int(NAcquisitions)))
-
-                for i in range(int(NAcquisitions)):
-                    total_run_time = 0.0
-                    psyllid_filename = filename+'_'+str(i)+'.egg'
-                    logger.info('Data will be written to {}/{}'.format(directory, psyllid_filename))
-                    if not os.path.exists(directory):
-                        os.makedirs(directory)
-                    # last acquisition
-                    if total_run_time > duration-max_duration:
-                        duration = duration-total_run_time
-                        logger.info('Going to tell psyllid to start the last run')
-                        payload = {'channel':self.channel_id, 'filename': os.path.join(directory, psyllid_filename), 'duration':duration}
+                # last acquisition
+                if total_run_time > duration-max_duration:
+                    duration = duration-total_run_time
+                    logger.info('Going to tell psyllid to start the last run')
+                    payload = {'channel':self.channel_id, 'filename': os.path.join(directory, psyllid_filename), 'duration':duration}
+                    try:
                         result = self.provider.cmd(self.psyllid_interface, 'start_run', payload=payload)
-                    # acquisitions with duration = max_duration
-                    else:
-                        total_run_time+=max_duration
-                        logger.info('Going to tell psyllid to start run')
-                        payload = {'channel':self.channel_id, 'filename': os.path.join(directory, psyllid_filename), 'duration':max_duration}
+                    except:
+                        logger.error('Starting Psyllid run failed')
+                        logger.info('unblock channel')
+                        payload = {'channel': self.channel_id}
+                        result = self.provider.cmd(self.daq_target, 'unblock_channel', payload=payload)
+                        raise core.exceptions.DriplineGenericDAQError('Starting Psyllid run failed')
+
+                # acquisitions with duration = max_duration
+                else:
+                    total_run_time+=max_duration
+                    logger.info('Going to tell psyllid to start run')
+                    payload = {'channel':self.channel_id, 'filename': os.path.join(directory, psyllid_filename), 'duration':max_duration}
+                    try:
                         result = self.provider.cmd(self.psyllid_interface, 'start_run', payload=payload)
-                    time.sleep(self._max_duration)
-        except:
-            logger.error('Starting Psyllid run failed')
-
+                    except:
```

Comment:
same exception handling comment, there is a big difference between "`psyllid_interface` returned an error" and "the local function call failed". We want to be sure not to hide the latter sort of errors.